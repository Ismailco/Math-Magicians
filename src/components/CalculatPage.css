.calculatPage {
  display: flex;
  justify-content: space-around;
  align-items: flex-start;
  margin-top: 50px;
  margin-left: 20px;
  margin-right: 20px;
}

.opening {
  font-weight: 600;
  font-size: 1.5em;
}

.calculator {
  border: 3px solid transparent;
  width: 500px;
}

@media screen and (max-width: 700px) {
  .calculatPage {
    flex-direction: column;
  }

  .calculator {
    margin-top: 70px;
  }
}
