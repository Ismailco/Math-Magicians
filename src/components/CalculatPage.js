import './CalculatPage.css';
import React from 'react';
import Calculator from './Calculator';

function CalculatPage() {
  return (
    <div className="calculatPage">
      <p className="paragraph">Let&apos;s do some math!</p>
      <Calculator className="calculator" />
    </div>
  );
}

export default CalculatPage;
