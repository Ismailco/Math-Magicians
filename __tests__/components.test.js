/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react';
import React from 'react';
import QuotesPage from '../src/components/Quotes';
import CalculatPage from '../src/components/CalculatPage';
import HomePage from '../src/components/Home';

describe('Use snapshots to test the components', () => {
  it('Create the Home page snapshot', () => {
    const { asFragment } = render(<HomePage />);
    expect(asFragment(<HomePage />)).toMatchSnapshot();
  });
  it('Create the Caculate page snapshot', () => {
    const { asFragment } = render(<CalculatPage />);
    expect(asFragment(<CalculatPage />)).toMatchSnapshot();
  });
  it('Create the Quote page snapshot', () => {
    const { asFragment } = render(<QuotesPage />);
    expect(asFragment(<QuotesPage />)).toMatchSnapshot();
  });
});
