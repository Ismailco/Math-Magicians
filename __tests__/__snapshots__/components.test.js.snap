// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Use snapshots to test the components Create the Caculate page snapshot 1`] = `
<DocumentFragment>
  <div
    class="calculatPage"
  >
    <p
      class="opening"
    >
      Let's do some math!
    </p>
    <div
      class="calculator"
    >
      <input
        class="input"
        readonly=""
        value="0"
      />
      <div
        class="test"
      >
        <div
          class="numbers-symbols"
        >
          <button
            class="buttons"
            type="button"
          >
            AC
          </button>
          <button
            class="buttons"
            type="button"
          >
            +/-
          </button>
          <button
            class="buttons"
            type="button"
          >
            %
          </button>
          <button
            class="buttons operators"
            type="button"
          >
            รท
          </button>
          <button
            class="buttons numbers"
            type="button"
          >
            7
          </button>
          <button
            class="buttons numbers"
            type="button"
          >
            8
          </button>
          <button
            class="buttons numbers"
            type="button"
          >
            9
          </button>
          <button
            class="buttons operators"
            type="button"
          >
            x
          </button>
          <button
            class="buttons numbers"
            type="button"
          >
            4
          </button>
          <button
            class="buttons numbers"
            type="button"
          >
            5
          </button>
          <button
            class="buttons numbers"
            type="button"
          >
            6
          </button>
          <button
            class="buttons operators"
            type="button"
          >
            -
          </button>
          <button
            class="buttons numbers"
            type="button"
          >
            1
          </button>
          <button
            class="buttons numbers"
            type="button"
          >
            2
          </button>
          <button
            class="buttons numbers"
            type="button"
          >
            3
          </button>
          <button
            class="buttons operators"
            type="button"
          >
            +
          </button>
          <button
            class="buttons numbers twoColl"
            type="button"
          >
            0
          </button>
          <button
            class="buttons"
            type="button"
          >
            .
          </button>
          <button
            class="buttons operators"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Use snapshots to test the components Create the Home page snapshot 1`] = `
<DocumentFragment>
  <div
    class="home-page"
  >
    <h2
      class="headline"
    >
      Welcome to our page!
    </h2>
    <p
      class="paragraph"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. At minima asperiores quisquam mollitia consectetur libero sunt perspiciatis, et sed sint ex in labore, aliquid magni laboriosam iure expedita. Fugit, amet.
    </p>
    <p
      class="paragraph"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Animi neque consectetur ut eveniet vel dignissimos et modi quibusdam impedit necessitatibus, dolores, asperiores debitis dolor dolorem soluta ipsa fugit iusto placeat?
    </p>
    <p
      class="paragraph"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolorum reprehenderit natus est et tempore, quisquam ratione expedita quam asperiores repellendus sint. Eveniet doloribus aliquam laudantium commodi, tempora odio repudiandae nemo?
    </p>
  </div>
</DocumentFragment>
`;

exports[`Use snapshots to test the components Create the Quote page snapshot 1`] = `
<DocumentFragment>
  <div
    class="quotes-page"
  >
    <h3
      class="quote"
    >
      Mathematics is not about numbers, equations, computations, or algorithms: it is about undrstanding. -William Paul Thurston
    </h3>
  </div>
</DocumentFragment>
`;
